packages/lab/src/combo-box-next/useComboBox.tsx

```diff
@@ -0,0 +1,136 @@
+import { FocusEvent, KeyboardEvent, SyntheticEvent } from "react";
+import { useList, UseListProps } from "../list-next/useList";
+import { useComboboxPortal, UseComboBoxPortalProps } from "./useComboboxPortal";
+
+interface UseComboBoxProps {
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+  onMouseOver?: (event: SyntheticEvent) => void;
+  PortalProps?: UseComboBoxPortalProps;
+  listProps: UseListProps;
+}
+
+export const useComboBox = ({
+  onFocus,
+  onBlur,
+  onMouseOver,
+  PortalProps,
+  listProps,
+}: UseComboBoxProps) => {
+  const {
+    open,
+    setOpen,
+    floating,
+    reference,
+    getPortalProps,
+    getTriggerProps,
+  } = useComboboxPortal(PortalProps);
+
+  const {
+    keyDownHandler: listKeyDownHandler,
+    focusHandler: listFocusHandler,
+    activeDescendant,
+    focusVisibleRef,
+    selectedItem,
+    setSelectedItem,
+    setHighlightedItem,
+    highlightedItem,
+  } = useList({
+    ...listProps,
+  });
+
+  const focusHandler = (event: FocusEvent<HTMLInputElement>) => {
+    setOpen(true);
+    listFocusHandler(event);
+    onFocus?.(event);
+  };
+
+  const blurHandler = (event: FocusEvent<HTMLInputElement>) => {
+    setOpen(false);
+    if (!selectedItem) {
+      setSelectedItem(undefined);
+      setHighlightedItem(undefined);
+    }
+    onBlur?.(event);
+  };
+
+  const mouseOverHandler = (event: SyntheticEvent<HTMLElement>) => {
+    setHighlightedItem(event.currentTarget.dataset.value);
+    onMouseOver?.(event);
+  };
+
+  const keyDownHandler = (event: KeyboardEvent<HTMLInputElement>) => {
+    const { key, altKey } = event;
+    switch (key) {
+      case "ArrowDown":
+      case "ArrowUp":
```

Comment:
one thing im not too sure about - for dropdown, i mimicked UITK dropdown behaviour where ArrowUp would not open list, only ArrowDown/Enter/Space would. Wondering if it's something to be discussed with Ben to finalise on whether list should open also on ArrowUp. Examples [here](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/) do open list on both ArrowUp/Down 

@bhoppers2008 what's your thoughts

Reply:
Interesting, agree, it's a bit odd we don't have the same behaviour for both...
for CB UITK both arrows open, 
```
If focus is in the input, and the list is closed, opens the list and moves focus to the first focusable item in the list.
```